step/src/main/java/com/google/sps/CalendarManager.java

```diff
@@ -0,0 +1,150 @@
+  }
+
+  // hostUser is the event owner and guestUser will receive an email invite
+  private static void pushMatchEvent(User hostUser, User guestUser, Event event) {
+    Calendar calendar = getCalendar(hostUser);
+    try {
+      String calendarId = "primary";
+      event = calendar.events().insert(calendarId, event).execute();
+      System.out.printf("Event created: %s\n", event.getHtmlLink());
+    } catch (Exception e) {
+      System.out.println("Unable to push match event.");
+      e.printStackTrace();
+    }
+  }
+
+  private static Calendar getCalendar(User user) {
+    return new Calendar.Builder(
+        new NetHttpTransport(),
+        new JacksonFactory(),
+        user.getCredential()
+      ).setApplicationName("Friend Matching Plus").build(); // TODO: not sure if app name matters
+  }
+
+}
+
+class MatchEventBuilder {
+  private Event matchEvent;
+
+  public MatchEventBuilder() {
+    matchEvent = new Event();
+  }
+
+  public MatchEventBuilder setAttendees(User hostUser, User guestUser) {
+    matchEvent.setSummary("FMP: " + hostUser.getName() + " / " + guestUser.getName());
+
+    matchEvent.setDescription(hostUser.getName() + " and " + guestUser.getName()
+                              + " have matched on Friend Matching Plus!");
+
+    matchEvent.setAttendees(Arrays.asList(new EventAttendee[] {
+      new EventAttendee().setEmail(guestUser.getEmail())
+    }));
+
+    return this;
+  }
+
+  // ZonedDateTime guide: https://www.baeldung.com/java-8-date-time-intro
+  public MatchEventBuilder setStartDateTime(int year, int month, int day, int hour, int minute) {
+    String dateTimeString = dateTimeString(year, month, day, hour, minute);
+
+    EventDateTime start = new EventDateTime()
+      .setDateTime(new DateTime(dateTimeString));
+      // .setTimeZone("America/Los_Angeles"); // TODO how to configure time zones?
+    matchEvent.setStart(start);
+
+    // Assume a 1 hour event.
+    DateTimeFormatter dateTimeFormatter = DateTimeFormatter.ISO_OFFSET_DATE_TIME;
+    ZonedDateTime zonedDateTime = ZonedDateTime.parse(dateTimeString, dateTimeFormatter);
+    zonedDateTime = zonedDateTime.plusHours(1);
+    dateTimeString = zonedDateTime.format(DateTimeFormatter.ISO_OFFSET_DATE_TIME);
+
+    EventDateTime end = new EventDateTime()
+      .setDateTime(new DateTime(dateTimeString));
+    matchEvent.setEnd(end);
+
+    return this;
+  }
+
+  public Event build() {
+    return matchEvent;
+  }
+
+  // createDateTimeString(2020, 07, 21, 9, 0, 0) -> "2020-07-21T09:00:00-05:00"
+  private static String dateTimeString(int year, int month, int day, int hour, int minute) {
+    int timeZone = -5; // central time zone. TODO(adamsamuelson): eliminate timezones
+
+    if (timeZone < 0) {
+      timeZone *= -1;
+      return String.format("%04d-%02d-%02dT%02d:%02d:%02d-%02d:%02d",
+                      year, month, day, hour, minute, 0, timeZone, 0);
+    } else {
+      return String.format("%04d-%02d-%02dT%02d:%02d:%02d+%02d:%02d",
```

Comment:
Duplicated code. You could just have the if without the else here and return String.format after since it happens in both cases.

Reply:
This isn't duplicate code, the difference in the format string is the "-" in the top format vs the "+" in the bottom format in the 10th to last character. I was having trouble getting %02d to format a negative int to 2 digits and this was the best solution I could come up with. Either way, I'm going to try to eliminate this method as Brenda suggested.